rotkehlchen/chain/ethereum/modules/thegraph/balances.py

```diff
@@ -0,0 +1,122 @@
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.accounting.structures.evm_event import EvmProduct
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.ethereum.interfaces.balances import ProtocolWithBalance
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.tokens import get_chunk_size_call_order
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.assets import A_GRT
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.fval import FVal
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+from .constants import CPT_THEGRAPH
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.interfaces.balances import BalancesType
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.types import CHAIN_IDS_WITH_BALANCE_PROTOCOLS
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class ThegraphBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            database: DBHandler,
+            evm_inquirer: 'EvmNodeInquirer',
+            chain_id: 'CHAIN_IDS_WITH_BALANCE_PROTOCOLS',
```

Comment:
I'm not sure if is possible since I don't know where this is caleld but maybe you can just say in the types ethereum? since is not valid for all the evm chains

Reply:
The Graph is only on Ethereum, yep.
You mean something like this?
```suggestion
     def __init__(
            self,
            database: DBHandler,
            evm_inquirer: 'EvmNodeInquirer',
            chain_id: 'Literal[ChainID.ETHEREUM]',
    ):
        super().__init__(
            database=database,
            evm_inquirer=evm_inquirer,
            chain_id=chain_id,
            counterparty=CPT_THEGRAPH,
        )
```

Reply:
yesm and can it be ethreum inquirer too?

Reply:
`evm_inquirer: 'EthereumInquirer'` works fine, but with `chain_id: 'Literal[ChainID.ETHEREUM]'` there's a linting error:
```bash
rotkehlchen/chain/aggregator.py:1036: error: Argument "chain_id" has incompatible type "Literal[ChainID.ETHEREUM, ChainID.OPTIMISM]"; expected "Literal[ChainID.ETHEREUM]"  [arg-type]
```
I wasn't expecting something from upstream... Lmk if it's worth to figure this one out.